From: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Subject: fix nfs service name to nfs-kernel-server

Upstream code used to comment in/out service script names related to a specific
distribution. Recently, they've changed this to a method of detecting on which
distribution ctdb is running, and then setting the systemd unit file name, for
example.

Because of this new detection mechanism, it is impossible to suggest upstream a
change like this patch and, at the same time, backporting this new code could
bring more problems. Instead, this patch should be kept until a new merge with
upstream is done.

This temporary patch fixes the NFS service name for Debian & Ubuntu.

Signed-off-by: Rafael David Tinoco <rafaeldtinoco@ubuntu.com>

Bug-Debian: https://bugs.debian.org/929931
Bug-Ubuntu: https://bugs.launchpad.net/bugs/722201
Last-Update: 2018-06-26
---
 ctdb/config/events/legacy/60.nfs.script | 4 ++--
 ctdb/config/nfs-linux-kernel-callout    | 3 +++
 ctdb/config/statd-callout               | 4 ++--
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/ctdb/config/events/legacy/60.nfs.script b/ctdb/config/events/legacy/60.nfs.script
index 2eb90b421c8..5c6a0903322 100755
--- a/ctdb/config/events/legacy/60.nfs.script
+++ b/ctdb/config/events/legacy/60.nfs.script
@@ -6,9 +6,9 @@
 
 . "${CTDB_BASE}/functions"
 
-service_name="nfs"
+service_name="nfs-kernel-server"
 
-load_system_config "nfs"
+load_system_config "nfs-kernel-server"
 
 load_script_options
 
diff --git a/ctdb/config/nfs-linux-kernel-callout b/ctdb/config/nfs-linux-kernel-callout
index 12ed17c6d9e..94111f10eb7 100755
--- a/ctdb/config/nfs-linux-kernel-callout
+++ b/ctdb/config/nfs-linux-kernel-callout
@@ -32,6 +32,9 @@ systemd-*)
 		: # Defaults only
 		;;
 	*-debian)
+		nfs_service="nfs-kernel-server"
+		nfs_lock_service=""
+		nfs_config="/etc/default/nfs-kernel-server"
 		nfs_rquotad_service="quotarpc"
 		;;
 	*)
diff --git a/ctdb/config/statd-callout b/ctdb/config/statd-callout
index b75135bbde5..db437a96d45 100755
--- a/ctdb/config/statd-callout
+++ b/ctdb/config/statd-callout
@@ -21,14 +21,14 @@ die ()
 }
 
 # Try different variables to find config file for NFS_HOSTNAME
-load_system_config "nfs" "nfs-common"
+load_system_config "nfs" "nfs-common" "nfs-kernel-server"
 
 [ -n "$NFS_HOSTNAME" ] || \
     die "NFS_HOSTNAME is not configured. statd-callout failed"
 
 ############################################################
 
-ctdb_setup_state_dir "service" "nfs"
+ctdb_setup_state_dir "service" "nfs-kernel-server"
 
 # script_state_dir set by ctdb_setup_state_dir()
 # shellcheck disable=SC2154
-- 
2.20.1

